libs/api-implementation/samples/src/lib/samples.controller.ts

```diff
@@ -0,0 +1,20 @@
+import {
+  Controller,
+  Post,
+  UseInterceptors,
+  UploadedFile,
+} from '@nestjs/common';
+import { FileInterceptor } from '@nestjs/platform-express';
+import { SamplesService } from './services/samples.service';
+
+@Controller('samples')
+export class SamplesController {
+  constructor(private service: SamplesService) {}
+
+  @Post('upload')
+  @UseInterceptors(FileInterceptor('file'))
+  uploadFile(@UploadedFile() file) {
+    this.service.writeToBucket(file);
+    return 200 
+  }
```

Comment:
I'm curious to understand how are we handling errors here? @zelzhan Can you please help me?

Reply:
Thank you for pointing out. Actually I've missed error handling here :). I've updated PR.

Reply:
Regarding the blacklisting and whitelisting: right now we are not checking the type of the upload. But yes, this is quite crucial. I'll create issues with the TODOs soon.

Reply:
Sounds good; thank you for clearing my doubts and taking my suggestions :D